### PR TITLE
[Resource] Rework mapping interfaces

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | -
| License         | MIT
| Doc PR          | -

 - [x] Made DoctrineTargetEntitiesResolver the CompilerPass
 - [x] Fixed for cases that does not use Doctrine ORM
 - [x] Reorganized PhpUnit tests in ResourceBundle
 - [x] Added tests for adding target entities

### CODE REVIEW

**Comment by gperdomor** [comment]
@pamil can you rebase please?

File: phpunit.xml.dist [hunk 1]
```diff
@@ -9,24 +9,7 @@
...
     <testsuites>
         <testsuite name="Sylius Test Suite">
-            <directory>src/*/Bundle/*Bundle/Tests</directory>
+            <directory>src/*/*/*/Tests</directory>
         </testsuite>
     </testsuites>
-
-    <php>
-        <server name="KERNEL_CLASS_PATH" value="/app/TestKernel.php" />
-        <server name="IS_DOCTRINE_ORM_SUPPORTED" value="true" />
```
Comment (gperdomor) [inline phpunit.xml.dist]: Why delete this?.

Comment (pamil) [inline phpunit.xml.dist]: There was some issue (fixed on master now), any modifications won't be needed if rebased to master. I'm going to have a look at the failing build tomorrow.

Comment (gperdomor) [inline phpunit.xml.dist]: Perfect, awesome job btw

**Comment by gperdomor** [comment]
ping @pamil :D

**Comment by pamil** [comment]
Took some time but finally it's working, I've opened Sylius/Sylius#4693 based on your PR as the base branch was modified.

**Comment by gperdomor** [comment]
@pamil don't worry... Thanks :+1:

**Pull request closed without merging**